### ISSUE TITLE
can't nest subgraphs in flowchart

### ISSUE THREAD

**Issue by masukomi** [knsv/mermaid#161]
The following is a valid subgraph

```
 graph TB
   subgraph Foo
     a1-->a2
   end
```

but this is not

```
 graph TB
   subgraph Foo
     a1-->a2
     subgraph Bar
       b1-->b2
     end
   end
```

I can't think of a reason subgraphs should not be nestable, and i can think of reasons I'd want them to be.

**Comment by knsv** [comment]
I agree that they should be nestable. Adding it to the list. Heavily involed in another project right now so the change will when I manage to find time.

**Comment by knsv** [comment]
The issue is fixed and will be included in next release.

**Comment by ubreddy** [comment]
looks like the fix has some bugs.

nested subgraph labels at 3 or levels deep are not working well, or if there are two subgraphs within a subgraph too.

**Comment by knsv** [comment]
Hmm. I wonder if there was a problem with the check in. This is my test page for nested subgraphs or perhaps this one is to simple.

![skarmavbild 2015-05-21 kl 18 49 26](https://cloud.githubusercontent.com/assets/5837277/7754112/41279f90-ffea-11e4-8caa-633e44d82b29.png)

**Comment by ubreddy** [comment]
<div class="mermaid">graph TB;  I --> IG((IG))    subgraph publicnet     IG --> ELB     NAT  end  subgraph webnet     ELB --> Web1     ELB --> Web2     Web1 --> elb2     Web2 --> elb2  end  subgraph appnet             subgraph C1        subgraph C1cluster1          elb2 --> C1Prod1Ins1          elb2 --> C1Prod1InsN        end        subgraph C1cluster2          elb2 --> C1Prod2Ins1          elb2 --> C1Prod2InsN        end        subgraph C1cluster3          elb2 --> C1Prod3Ins1          elb2 --> C1Prod3InsN        end     end     subgraph C2        subgraph C2cluster1          elb2 --> C2Prod1Ins1          elb2 --> C2Prod1InsN        end        subgraph C2cluster2          elb2 --> C2Prod2Ins1          elb2 --> C2Prod2InsN        end        subgraph C2cluster3          elb2 --> C2Prod3Ins1          elb2 --> C2Prod3InsN        end     end  end  subgraph dbnet    C1Prod1Ins1 --> RDS1    C1Prod1InsN --> RDS1    C1Prod2Ins1 --> RDS1    C1Prod2Ins2 --> RDS1    C1Prod3Ins1 --> RDS1    C1Prod3Ins2 --> RDS1    C2Prod1Ins1 --> RDS2     C2Prod1InsN --> RDS2    C2Prod2Ins1 --> RDS2     C2Prod2Ins2 --> RDS2    C2Prod3Ins1 --> RDS2    C2Prod3Ins2 --> RDS2  end


</div> e.g I tried this... I notice graph label is being changed ( C1 and C1Cluster3)... and if I change the orientation of graph to LR it goes haywire.
Uday 


     On Thursday, 21 May 2015 10:21 PM, Knut Sveidqvist <notifications@github.com> wrote:
   

 Hmm. I wonder if there was a problem with the check in. This is my test page for nested subgraphs or perhaps this one is to simple.—

**Comment by knsv** [comment]
Thanks for your example. I can replicate the issue now.

**Comment by knsv** [comment]
The issue was a bit harder ti fix then anticipated. The proper way to fix this involves a change in dagre-d3. I have submitted such a pull request and if/when this is included in dagre-d3 I will finish the solution of this issue.

The problem is to identify the area in svg that is a subgraph. The positioning engine dagre-d3 positions thins for a good layout that does not necessarily correspond to the order things are in the graph definition so it is a guessing game to find the right section.

**Comment by knsv** [comment]
The change in dagre-d3 has been released and the fix of the issue with the crazy labels has been pushed into (dev-0.5.0). In that branch there are also some experimental changes for packaging mermaid and for integration so use with care. Look at test/web_render.html for an example of nested subgraphs.

**Comment by knsv** [comment]
A fix for this has been released in version 0.5.0

**Comment by ubreddy** [comment]
@Knut
Replaced mermaid.full.js of 0.4 with mermaid.js of version 0.5 and the diagrams dont render? read the docs... It should have worked. 
What am I missing? any guidance?
Regards Uday 


     On Sunday, 7 June 2015 9:04 PM, Knut Sveidqvist <notifications@github.com> wrote:
   

 A fix for this has been released in version 0.5.0—

**Comment by knsv** [comment]
Thanks for pointing this out. You also need to add the code below to your page. I will update the docs.
<script>mermaid.initialize({startOnLoad:true});</script>



--  
Knut Sveidqvist
Sent with Sparrow (http://www.sparrowmailapp.com/?sig)
